cpp/tests/unit_tests/kernels/mixtureOfExpertsOssTest.cu

```diff
@@ -0,0 +1,2037 @@
+                    << "Incorrect final value at for token: " << token_id << " offset: " << hidden_id
+                    << " hidden_size: " << mHiddenSize << " inter_size: " << mInterSize;
+            }
+        }
+    }
+
+    void BasicPermuteTest(
+        int k = 1, int64_t hidden_size = DEFAULT_HIDDEN_SIZE, int64_t num_experts = 4, int64_t num_tokens = 3);
+
+    std::vector<int> calcPermuteMapExpertParallel(std::vector<int> const& expected_experts);
+
+    void ExpertParallelTest(int k = 1, int64_t hidden_size = DEFAULT_HIDDEN_SIZE, int64_t num_experts = 4,
+        int64_t num_tokens = 3, float inter_size_fraction = 4.0f)
+    {
+        mInterSizeFraction = inter_size_fraction;
+        // 2 experts per rank
+        ParallelismTest(k, 1, num_experts / 2, hidden_size, num_experts, num_tokens);
+        // 1 expert per rank
+        ParallelismTest(k, 1, num_experts, hidden_size, num_experts, num_tokens);
+    }
+
+    // Tensor parallel tests default to inter_size_fraction = 1.0f so that all ranks have interesting values
+    void TensorParallelTest(int k = 1, int64_t hidden_size = DEFAULT_HIDDEN_SIZE, int64_t num_experts = 4,
+        int64_t num_tokens = 3, float inter_size_fraction = 1.0f)
+    {
+        mInterSizeFraction = inter_size_fraction;
+        ParallelismTest(k, 2, 1, hidden_size, num_experts, num_tokens);
+        ParallelismTest(k, 4, 1, hidden_size, num_experts, num_tokens);
+        ParallelismTest(k, 8, 1, hidden_size, num_experts, num_tokens);
+    }
+
+    void MixedParallelTest(int k = 1, int64_t hidden_size = DEFAULT_HIDDEN_SIZE, int64_t num_experts = 4,
+        int64_t num_tokens = 3, float inter_size_fraction = 1.0f)
+    {
+        mInterSizeFraction = inter_size_fraction;
+
+        // 2 experts per rank
+        ParallelismTest(k, 2, num_experts / 2, hidden_size, num_experts, num_tokens);
+        ParallelismTest(k, 8, num_experts / 2, hidden_size, num_experts, num_tokens);
+
+        // 1 expert per rank
+        ParallelismTest(k, 2, num_experts, hidden_size, num_experts, num_tokens);
+        ParallelismTest(k, 8, num_experts, hidden_size, num_experts, num_tokens);
+    }
+
+    void ParallelismTest(int k = 1, int tp_size = 4, int ep_size = 2, int64_t hidden_size = DEFAULT_HIDDEN_SIZE,
+        int64_t num_experts = 4, int64_t num_tokens = 3);
+};
+
+template <class WeightParams>
+using LargeMixtureOfExpertsTest = MixtureOfExpertsTest<WeightParams>;
+
+template <class DataType_, class WeightType_ = DataType_, class OutputType_ = DataType_>
+struct WeightParams
+{
+    using DataType = DataType_;
+    using WeightType = WeightType_;
+    using OutputType = OutputType_;
+};
+
+// TODO Fix int quantized
+using Types = ::testing::Types<
+#ifdef ENABLE_BF16
+    WeightParams<__nv_bfloat16>,
+#endif
+#ifdef ENABLE_FP8
+    WeightParams<SafeFP8, SafeFP8, half>,
+#endif
+#ifdef ENABLE_FP4
+    WeightParams<SafeFP4, SafeFP4, half>,
+#endif
+    WeightParams<half>, WeightParams<float>
```

Comment:
I am guessing we are duplicating this because we dont want MXFP4. 
Instead of duplicating, please just add an `#ifdef` around adding it to this list in the existing moe test.

If we also need to test both compile definitions we should compile the same file twice with different compile definitions